pkgs/development/python-modules/snug/default.nix

```diff
@@ -0,0 +1,35 @@
+{ buildPythonPackage, lib, fetchFromGitHub, glibcLocales
+, pytest, pytest-mock, gentools
+, typing, singledispatch
+, isPy27, isPy33, isPy34
+}:
+
+buildPythonPackage rec {
+  pname = "snug";
+  version = "1.3.4";
+
+  src = fetchFromGitHub {
+    owner = "ariebovenberg";
+    repo = "snug";
+    rev = "v${version}";
+    sha256 = "0jmg0sivz9ljazlnsrrqaizrb3r7asy5pa0dj3idx49gbig4589i";
+  };
+
+  buildInputs = [ glibcLocales ];
+  LC_ALL = "en_US.UTF-8";
+
+  propagatedBuildInputs =
+    lib.optionals (isPy27 || isPy33) [ singledispatch ] ++
```

Comment:
```nix
lib.optionals (pythonOlder "3.4") [ singledispatch ] ++
lib.optionals (pythonOlder "3.5") [ typing ];
```

Using [`pythonOlder`](https://github.com/NixOS/nixpkgs/blob/6dbf15d627ecd1a18bf965000c86b63b70629ac6/pkgs/top-level/python-packages.nix#L22) feels nicer here.

Reply:
Absolutely. I didn't know that, thanks.